--- mlx5_flow.c.orig	2019-12-10 22:02:18.438477190 +0000
+++ mlx5_flow.c	2019-12-10 22:02:37.998316618 +0000
@@ -1484,6 +1484,7 @@
 						  " environment");
 		}
 	}
+#ifndef TREX_PATCH
 	if (spec) {
 		vlan_tag = spec->tci;
 		vlan_tag &= mask->tci;
@@ -1497,6 +1498,7 @@
 					  RTE_FLOW_ERROR_TYPE_ITEM_SPEC,
 					  item->spec,
 					  "VLAN cannot be empty");
+#endif
 	return 0;
 }
 
@@ -5235,7 +5237,7 @@
 	case RTE_ETH_FILTER_FDIR:
 		return flow_fdir_ctrl_func(dev, filter_op, arg);
 	default:
-		DRV_LOG(ERR, "port %u filter type (%d) not supported",
+		DRV_LOG(DEBUG, "port %u filter type (%d) not supported",
 			dev->data->port_id, filter_type);
 		rte_errno = ENOTSUP;
 		return -rte_errno;
